mwviews/api/pageviews.py

```diff
@@ -301,6 +303,120 @@ def top_articles(
+            month : int
+                default : yesterday's month
+            limit : int
+                limit the number of articles returned to only the top <limit>
+                default : 1000
+
+        :Returns:
+            a sorted list of countries that looks like: [
+                {
+                    rank: <int>,
+                    country: <str>,
+                    views: <str>,
+                    views_ceil: <int>
+                }
+                ...
+            ]
+        """
+        lastmonth = date.today().replace(day=1) - timedelta(days=1)
+        year = str(year or lastmonth.year)
+        month = str(month or lastmonth.month).rjust(2, '0')
+
+        url = '/'.join([endpoints['top-by-country'], project, access, year, month])
+
+        try:
+            result = requests.get(url, headers=self.headers).json()
+
+            if 'items' in result and len(result['items']) == 1:
+                r = result['items'][0]['countries']
+                r.sort(key=lambda x: x['rank'])
+                return r[0:(limit)]
+        except:
+            print('ERROR while fetching or parsing ' + url)
+            traceback.print_exc()
+            raise
+
+        raise Exception(
+            'The pageview API returned nothing useful at: {}'.format(url)
+        )
+
+    def top_articles_per_country(
+            self, country, access='all-access',
+            year=None, month=None, day=None, limit=1000):
+        """
+        Get top-1000 articles with highest pageview counts for a country (across all projects)
+        See `<https://wikimedia.org/api/rest_v1/metrics/pageviews/?doc\\
+                #!/Pageviews_data/get_metrics_pageviews_top_project\\
+                _access_year_month_day>`_
+
+        :Parameters:
+            country : str
+                an ISO 3166-1 alpha-2 code of a country such as FR (France) or IN (India)
+            access : str
+                access method (desktop, mobile-web, mobile-app, or by default, all-access)
+            year : int
+                default : yesterday's year
+            month : int
+                default : yesterday's month
+            day : int
+                default : yesterday's day
+            limit : int
+                limit the number of articles returned to only the top <limit>
+                default : 1000
+
+        :Returns:
+            a sorted list of articles that looks like: [
+                {
+                    rank: <int>,
+                    project: <str>,
+                    article: <str>,
+                    views_ceil: <int>
+                }
+                ...
+            ]
+        """
+        yesterday = date.today() - timedelta(days=1)
+        year = str(year or yesterday.year)
+        month = str(month or yesterday.month).rjust(2, '0')
+        day = str(day or yesterday.day).rjust(2, '0')
+        country = country.upper()  # common issue is lower-case country codes
```

Comment:
nice catch!

Reply:
thanks! where "common issue" = I did it :)

Reply:
oh, me too :)